tests/private_beam_test.py

```diff
@@ -0,0 +1,169 @@
+            # Arrange
+            pcol = pipeline | 'Create produce' >> beam.Create(
+                [1, 2, 3, 4, 5, 6])
+            budget_accountant = budget_accounting.NaiveBudgetAccountant(
+                total_epsilon=1, total_delta=0.01)
+            private_collection = (
+                pcol | 'Create private collection' >> private_beam.MakePrivate(
+                    budget_accountant=budget_accountant,
+                    privacy_id_extractor=PrivateBeamTest.privacy_id_extractor))
+
+            # Act and Assert
+            with self.assertRaises(TypeError) as context:
+                (private_collection | 'Non private transform on '
+                 'privateCollection' >> beam.Map(lambda x: x))
+            self.assertIsInstance(private_collection,
+                                  private_beam.PrivateCollection)
+            self.assertTrue(
+                "private_transform should be of type "
+                "PrivateTransform but is " in str(context.exception))
+
+    def test_transform_with_make_private_enabled_returns_private_collection(
+            self):
+        runner = fn_api_runner.FnApiRunner()
+        with beam.Pipeline(runner=runner) as pipeline:
+            # Arrange
+            pcol = pipeline | 'Create produce' >> beam.Create(
+                [1, 2, 3, 4, 5, 6])
+            budget_accountant = budget_accounting.NaiveBudgetAccountant(
+                total_epsilon=1, total_delta=0.01)
+            private_collection = (
+                pcol | 'Create private collection' >> private_beam.MakePrivate(
+                    budget_accountant=budget_accountant,
+                    privacy_id_extractor=PrivateBeamTest.privacy_id_extractor))
+
+            # Act
+            private_transformer = SimplePrivateTransform()
+            private_transformer.set_return_private()
+
+            transformed = private_collection | private_transformer
+
+            # Assert
+            self.assertIsInstance(transformed, private_beam.PrivateCollection)
+
+    def test_transform_with_make_private_disabled_returns_PCollection(self):
+        runner = fn_api_runner.FnApiRunner()
+        with beam.Pipeline(runner=runner) as pipeline:
+            # Arrange
+            pcol = pipeline | 'Create produce' >> beam.Create(
+                [1, 2, 3, 4, 5, 6])
+            budget_accountant = budget_accounting.NaiveBudgetAccountant(
+                total_epsilon=1, total_delta=0.01)
+            private_collection = (
+                pcol | 'Create private collection' >> private_beam.MakePrivate(
+                    budget_accountant=budget_accountant,
+                    privacy_id_extractor=PrivateBeamTest.privacy_id_extractor))
+
+            # Act
+            transformed = private_collection | SimplePrivateTransform()
+
+            # Assert
+            self.assertIsInstance(transformed, PCollection)
+
+    @patch('pipeline_dp.dp_engine.DPEngine.aggregate')
```

Comment:
Nice, that's the right way to test, to ensure that DPEngine is called in correct way!

Reply:
Thanks!